File: src/telegram/style.py
from colorama import Fore, Back, Style
import os


class LogStyle:
    def error(self, msg): return (Fore.RED + f"[!] {msg}" + Fore.RESET) 
    def sysinfo(self, msg): return (Fore.MAGENTA + f"[i] {msg}" + Fore.RESET)
    def success(self, msg): return (Fore.GREEN + f"[!] {msg}" + Fore.RESET)
    def info(self, msg): return (Fore.BLUE + f"\t[+] {msg}" + Fore.RESET)
    def line(self): return ('─' * os.get_terminal_size().columns)
    def solidLine(self): return (Back.WHITE + '⠀' * os.get_terminal_size().columns + '\n' + Back.RESET)
    def data(self, msg): return (Fore.YELLOW + str(msg) + Fore.RESET)
    def subprocess(self, msg): return (Fore.CYAN + '\n' + str(msg) + Fore.RESET)

style = LogStyle()

File: src/modules/style.ts
/**
 * @module style
 * @description Utility module to keep the import for style. Keeping style consistent among the other modules
 * @category misc
 */

import chalk from "chalk";


/**
 * @constant style
 *
 * @type {{ error: (msg: unknown) => any; sysinfo: (msg: string) => any; success: (msg: string) => any; info: (msg: string) => any; line: () => any; solidLine: () => any; data: (msg: string | number) => any; subprocess: (msg: string) => any; }}
 */
export const style = {
  error: (msg : string | null | undefined | unknown) => chalk.red("[!] " + msg),
  sysinfo: (msg : string | null | undefined) => chalk.magenta("[i] " + msg),
  success: (msg : string | null | undefined) => chalk.green("[!] " + msg),
  info: (msg : string | null | undefined) => chalk.blue("\t[+] " + msg),
  line: () => chalk.white('─'.repeat(process.stdout.columns)),
  solidLine: () => chalk.bgWhite('⠀'.repeat(process.stdout.columns) + '\n'),
  data: (msg : string | number | null | undefined) => chalk.yellow(msg),
  subprocess: (msg : string | null | undefined) => chalk.cyan("\n" + msg + "\n"),
};